### DISCUSSION TITLE
Codepage planning

### DISCUSSION THREAD

**Discussion by lyxal** [Vyxal/Vyxal#715] [Version 3]
Current codepage (feel free to edit with ideas):

```
  0123456789ABCDEF
  ----------------
0|¡¿¤¢€©®§¶æð▲▼ƒ⟨⟩
1|λƛΩΛµ∆øÞŒʑʒʓ×÷£¥
2| !"#$%&'()*+,-./
3|0123456789:;<=>?
4|@ABCDEFGHIJKLMNO
5|PQRSTUVWXYZ[\]^_
6|`abcdefghijklmno
7|pqrstuvwxyz{|}~¦
8|ȦḂĊḊĖḞĠḢİĿṀṄȮṖṘṠ
9|ṪẆẊẎŻȧḃċḋėḟġḣŀṁṅ
A|ȯṗṙṡṫẇẋẏż≤≥≠⁼→←‡
B|₀₁₂₃₄₅₆₇₈₉«»⌐₩₳⊻
C|⁰¹²³⁴⁵⁶⁷⁸⁹⁺⁻⁾√∑Π
D|ẠḄḌẸḤỊḲḶṂṆỌṚṢṬạḅ
E|ḍẹịḷṃọṛṣṭ±□‛“¯≈ı
F|‹›βεδζξστχω↑↓∥∦≡
```

(`.` represents an unfilled slot)
(the `¶` at `0A` is a newline)
(the `.` at `2E` is a literal `.`)

Respond with character ranges you think should be drawn from, or aesthetic ideas/guidelines you want to see followed.

**Comment by cgccuser** [comment]
I assume all the printable ASCII characters would be in their normal spots.

**Reply by lyxal** [reply]
I'll be adding those in shortly

**Comment by cgccuser** [comment]
We should try to make sure the characters are all easily distinguishable and can be attached to some set of builtins with mnemonics.

**Comment by lyxal** [comment]
We should try to make sure all characters are the same width

**Comment by lyxal** [comment]
We should try to make sure there aren't any homoglyphs or characters that might be easily confused.

**Comment by razetime** [comment]
Replace newline with `¶` or some other less ambiguous replacement?

**Reply by razetime** [reply]
display purposes only

**Comment by AMiller42** [comment]
~Fill the remaining slots with unprintables~

**Reply by mathcat4** [reply]
No, let's invent a new encoding, in which all characters are rendered as "LOL".

**Comment by zoomlogo** [comment]
Subscript / Superscript number.

IDK ok this is an idea.

**Comment by lyxal** [comment]
**Modifiers**: `¡¿¤¢€©®§æðƒ)/\]v}~¦βεδζξστχω↑↓∥∦≡`

**Lambdas**: `λƛΩΛµ`

**Digraphs**: `∆øÞŒ`

**Ranges**:  `zʑʒʓ`

**Compressed strings**: `▲▼`

**Global Array Stuff**" `⁺⁻⁾`

**Reply by mathcat4** [reply]
Why not use an uppercase greek lambda (`Λ`) for lambdas?

**Comment by pxeger** [comment]
Did you mean to exclude <code>&#96;</code> from the codepage, making it not compatible with ASCII?

**Reply by AMiller42** [reply]
I doubt it was intentional, so I fixed it
